### PR TITLE
[MM-50347] fix: Make desktop app landing page optional

### PR DESCRIPTION
**Pull request by kiranbabu7**
#### Summary

- Added  `Enable Desktop App Landing Page` to `System Console -> SITE CONFIGURATION - > Customization`
`Enable Desktop App Landing Page` value is by default `true` which shows `Landing Page` (present working)
- If set to `false`, `Landing Page` won't be shown to Web Users

#### Fixes:
Github https://github.com/mattermost/mattermost/issues/23484
Jira https://mattermost.atlassian.net/browse/MM-50347
#### Screenshots

|  before  | 
image
| after  |
image

#### Release Note

```release-note
- Added a new config setting ServiceSettings.EnableDesktopLandingPage.
- Added `Enable Desktop App Landing Page` to `System Console -> SITE CONFIGURATION - > Customization`
`Enable Desktop App Landing Page` value is by default `true` .
 If set to `false`, `Landing Page` won't be shown to Web Users
```

### CODE REVIEW

File: webapp/platform/types/src/config.ts [hunk 1]
```diff
@@ -64,6 +64,7 @@ export type ClientConfig = {
     EnableCustomTermsOfService: string;
     EnableDeveloper: string;
     EnableDiagnostics: string;
+    EnableDesktopLandingPage: 'true' | 'false';
```
Comment (devinbinnie) [inline webapp/platform/types/src/config.ts:67]: This should be `string` to conform with the other settings

**Review by devinbinnie — COMMENTED** [review]
THanks @kiranbabu7, code looks good but it seems though we have some lint/type/tests broken, as well as missing i18n strings. Can you run those checks and update as necessary?

It might also be necessary to merge master into your branch to get the latest changes.

**Comment by hmhealey** [comment]
@kiranbabu7 Thanks for your work on this! We've actually found that we need this feature to set up some automated testing which is currently blocked by the landing page, so I need to take this ticket over. I'll try to build off your changes so that you should still get credit for them though. We definitely appreciate the time and effort you've put in on this still!

**Comment by kiranbabu7** [comment]
@hmhealey thanks for taking this over I have some issue with my laptop due to which I am unable to run lint in local

**Comment by hmhealey** [comment]
Oh yeah, I forgot to do that. Thanks for the reminder! I'm sure the docs team really appreciates it 😅

**Pull request closed without merging**